src/btree/multi.ts

```diff
@@ -0,0 +1,78 @@
+import { RangeResolver } from "../resolver";
+import { MemoryPointer } from "./node";
+
+export class LinkedMetaPage {
+	private resolver: RangeResolver;
+	private offset: number;
+
+	constructor(resolver: RangeResolver, offset: number) {
+		this.resolver = resolver;
+		this.offset = offset;
+	}
+
+	async root(): Promise<MemoryPointer | null> {
+		// we seek by 12 bytes since offset is 8 bytes, length is 4 bytes
+		const { data } = await this.resolver({
+			start: this.offset,
+			end: this.offset + 12,
+		});
+
+		if (data.byteLength !== 12) {
+			return null;
+		}
+
+		const view = new DataView(data);
+
+		const pointerOffset = view.getBigUint64(0);
+		const lengthOffset = view.getUint32(8);
+
+		return {
+			offset: pointerOffset,
+			length: lengthOffset,
+		};
+	}
+
+	async metadata(): Promise<ArrayBuffer> {
+		const { data: lengthData } = await this.resolver({
+			start: this.offset + 24,
+			end: this.offset + 28,
```

Comment:
It's reasonable to assume that the metadata is the remainder of the page, so the resolver only needs to be called once. That is, `start: this.offset + 24, end: this.offset + 4096 - 1`.

The first four bytes represent the length and then discard the remaining data.

Reply:
> The first four bytes represent the length and then discard the remaining data.

I assume you mean retrieve the length from the first four bytes, then retrieve the metadata from:

```js
{ start: this.offset + 28, end: this.offset + 28 + metdatalength - 1 }
```

Discard the rest

Reply:
Ah yes, however I'm saying that you can read it in one go. So I think in line with the other comment the idea is that when pulling a `LinkedMetaPage`, read all 4096 bytes at once then slice it up as you wish instead of calling the resolver multiple times for each desired range.

There's a slight benefit to doing it this way, namely the backend right now guarantees that each page is written atomically, so we want to read in pages too to guarantee we're not desynchronizing across different requests.